libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1815,6 +1846,19 @@ void GCS_MAVLINK::handle_timesync(mavlink_message_t *msg)
         );
 }
 
+/*
+ * broadcast a timesync message.  We may get multiple responses to this request.
+ */
+void GCS_MAVLINK::send_timesync()
+{
+    _timesync_request.sent_ts1 = timesync_timestamp_ns();
```

Comment:
this is used for sending a TIMESYNC on a timer, not in response to an incoming packet. The timesync_timestamp_ns() looks at the last packet arrival time, which won't have been a TIMESYNC. This means it is adding time jitter, not removing it